controller/share_manager_controller.go

```diff
@@ -438,10 +435,10 @@ func (c *ShareManagerController) syncShareManagerVolume(sm *longhorn.ShareManage
 		// ensure volume gets detached if share manager needs to be cleaned up and hasn't stopped yet
 		// we need the isNotNeeded var so we don't accidentally detach manually attached volumes,
 		// while the share manager is no longer running (only run cleanup once)
-		if isNotNeeded && sm.Status.State != types.ShareManagerStateStopped {
+		if isNotNeeded && sm.Status.State != types.ShareManagerStateStopping && sm.Status.State != types.ShareManagerStateStopped {
```

Comment:
NIT: If `sm.Status.State != types.ShareManagerStateStopping && sm.Status.State != types.ShareManagerStateStopped` is used here, there is no need to re-check it below when setting `isNotNeeded` to `true`. Or vice versa.

Furthermore, I am thinking if we really need to check the share manager state before detachment. Is this check for avoiding detaching a stopped volume?

Reply:
Correct, this is so that if the share-manager is stopped we don't mess with the volume.
The extra state check is a little redundant but it helps make it more clear in this case.